### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by Cdm2883** [Monkopedia/kplusplus#1]
The reason seems to be that krapper_gen uses ksrpc to communicate with libclang and ksrpc uses ktor-server to communicate, but the native implementation of ktor-server doesn't support windows.

<details>

<summary>This is an error:</summary>

```
A problem occurred configuring project ':plugin'.
> Could not resolve all files for configuration ':plugin:classpath'.
   > Could not find kplusplus:krapper_gen:0.1.0.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/kplusplus/krapper_gen/0.1.0/krapper_gen-0.1.0.pom
       - https://plugins.gradle.org/m2/kplusplus/krapper_gen/0.1.0/krapper_gen-0.1.0.pom
     Required by:
         project :plugin > com.monkopedia.kplusplus.plugin:com.monkopedia.kplusplus.plugin.gradle.plugin:0.1.0 > com.monkopedia.kplusplus:plugin:0.1.0

Possible solution:
 - Declare repository providing the artifact, see the documentation at https://docs.gradle.org/current/userguide/declaring_repositories.html
```

</details>

<details>

<summary>This is the result of a composite build using git's submodule and then gradle:</summary>

```
A problem occurred configuring project ':plugin'.
> Could not determine the dependencies of task ':kplusplus:krapper_gen:linkDebugExecutableNative'.
   > Could not resolve all task dependencies for configuration ':kplusplus:krapper_gen:nativeCompileKlibraries'.
      > Could not resolve com.monkopedia:ksrpc:0.6.0.
        Required by:
            project :kplusplus:krapper_gen
         > No matching variant of com.monkopedia:ksrpc:0.6.0 was found. The consumer was configured to find a library for use during 'kotlin-api', preferably optimized for non-jvm, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native', attribute 'org.jetbrains.kotlin.native.target' with value 'mingw_x64' but:
             - Variant 'jsApiElements-published' capability com.monkopedia:ksrpc:0.6.0 declares a library for use during 'kotlin-api':
                 - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                     - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'mingw_x64')
             - Variant 'jsRuntimeElements-published' capability com.monkopedia:ksrpc:0.6.0 declares a library:
                 - Incompatible because this component declares a component for use during 'kotlin-runtime', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                     - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'mingw_x64')
             - Variant 'jvmApiElements-published' capability com.monkopedia:ksrpc:0.6.0 declares a library for use during compile-time:
                 - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                     - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'mingw_x64')
             - Variant 'jvmRuntimeElements-published' capability com.monkopedia:ksrpc:0.6.0 declares a library for use during runtime:
                 - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                     - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'mingw_x64')
             - Variant 'metadataApiElements' capability com.monkopedia:ksrpc:0.6.0 declares a library:
                 - Incompatible because this component declares a component for use during 'kotlin-metadata', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'common' and the consumer needed a component for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)
                     - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'mingw_x64')
             - Variant 'nativeApiElements-published' capability com.monkopedia:ksrpc:0.6.0 declares a library for use during 'kotlin-api', as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
                 - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'mingw_x64'
                 - Other compatible attribute:
                     - Doesn't say anything about its target Java environment (preferred optimized for non-jvm)

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
```
full version:
[errer.log](https://github.com/user-attachments/files/16454066/errer.log)

</details>

**Comment by monkopedia-coder** [comment]
Triage: labeling `blocked`. Windows (mingwX64) support is gated on an upstream dependency limitation — krapper_gen talks to libclang over ksrpc, and ksrpc's transport relies on ktor-server, whose Kotlin/Native implementation does not support Windows. The dependency-resolution errors here are a symptom of that missing mingw_x64 variant. This can't be resolved mechanically in this repo; it depends on upstream ktor-server / ksrpc gaining Windows native support. Bumping ksrpc here as a side effect is out of scope per release/consumer-impact policy.
